From 5058b09996181fbd398c799eeba6a1d83b083186 Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Tue, 11 Dec 2018 16:14:11 +0200
Subject: [PATCH 3/3] Make introspection support optional

Upstream has it optional again with commit 200a5869b5c1dc8 as well,
but that's included only since nautilus-3.29.90
---
 libnautilus-extension/meson.build | 2 ++
 meson_options.txt                 | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/libnautilus-extension/meson.build b/libnautilus-extension/meson.build
index 067ad2414..4702964a9 100644
--- a/libnautilus-extension/meson.build
+++ b/libnautilus-extension/meson.build
@@ -41,12 +41,14 @@ libnautilus_extension = shared_library ('nautilus-extension',
                                         version: nautilus_extension_version,
                                         install: true)
 
+if get_option('introspection')
 gnome.generate_gir (libnautilus_extension,
                     sources: libnautilus_extension_headers + libnautilus_extension_sources,
                     nsversion: '3.0',
                     namespace: 'Nautilus',
                     includes: ['Gtk-3.0', 'Gio-2.0', 'GLib-2.0'],
                     install: true)
+endif
 
 nautilus_extension = declare_dependency (link_with: libnautilus_extension,
                                          dependencies: libnautilus_extension_deps,
diff --git a/meson_options.txt b/meson_options.txt
index 0c23c7921..6ad3a72de 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,6 +1,10 @@
 option ('enable-profiling',
         type: 'boolean',
         value: false)
+option ('introspection',
+        type: 'boolean',
+        value: true,
+        description: 'Build GObject introspection data for extension library')
 option ('enable-nst-extension',
         type: 'boolean',
         value: true)
-- 
2.17.0

